src/domain/chains/chains.repository.ts

```diff
@@ -47,6 +48,28 @@ export class ChainsRepository implements IChainsRepository {
     return valid;
   }
 
+  async getAllChains(): Promise<Array<Chain>> {
+    const chains: Array<Chain> = [];
+
+    let offset: number | undefined;
+
+    while (true) {
+      const result = await this.getChains(undefined, offset);
+      chains.push(...result.results);
+
+      if (!result.next) {
+        break;
+      }
+
+      const url = new URL(result.next);
+      const paginationData = PaginationData.fromLimitAndOffset(url);
+      offset ??= 0;
+      offset += paginationData.offset;
```

Comment:
I think this is not correct. Considering for example 10 items per page (limit), it would work for the second page (offset was 0, + 10 = 10) and also for the third (previous offset is 10, + 10 = 20), but for the fourth it would take 20 as previous offset and the next URL would contain `?offset=30`, right? So it would set `50` as new offset. 

Maybe my math is wrong but I think we need to set the offset directly to the one extracted from the URL.

```typescript
offset = paginationData.offset;
```

Reply:
Good point. The `offset` is indeed "fixed". I change it in 16fcf76322f95771485d14cb8100fec7de82649b.